ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/task/WaitForPeerTask.java

```diff
@@ -40,14 +42,18 @@ public static WaitForPeerTask create(
 
   @Override
   protected void executeTask() {
+    final Random rand = new Random();
```

Comment:
It looks like you're using the random number to match up the waiting events with finished ...don't forget that with random you never know what you're going to get (well, there's a degree of determinism with `java.util.Random`), but be aware you could end up with duplicate values ...alternatively you could instance a static class counter 😁 

```
private final static AtomicLong EVENT_ID = new AtomicLong():
...

final long thisFuture = EVENT_ID.getAndIncrement();
...
```